From 1115fa3245146c5e1badc605e5bcf0555a68e6d3 Mon Sep 17 00:00:00 2001
From: Veaceslav Falico <vfalico@redhat.com>
Date: Thu, 21 Aug 2014 11:28:30 -0400
Subject: [ethernet] netxen: Miscellaneous conversions to ETH_ALEN

Message-id: <1408620515-703-6-git-send-email-vfalico@redhat.com>
Patchwork-id: 88516
O-Subject: [PATCH RHEL7 05/10] net:drivers/net: Miscellaneous conversions to ETH_ALEN
Bugzilla: 1089356
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

RHBZ#1089356

Testing:
Locally and by QLogic (which supplied the patch)

Description:
Backport of the following commit:

commit d458cdf712e0c671e8e819abb16ecd6e44f9daec
Author: Joe Perches <joe@perches.com>
Date:   Tue Oct 1 19:04:40 2013 -0700

    net:drivers/net: Miscellaneous conversions to ETH_ALEN

    Convert the memset/memcpy uses of 6 to ETH_ALEN
    where appropriate.

    Also convert some struct definitions and u8 array
    declarations of [6] to ETH_ALEN.

    Signed-off-by: Joe Perches <joe@perches.com>
    Acked-by: Arend van Spriel <arend@broadcom.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c b/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
index 8375cbd..67efe75 100644
--- a/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
+++ b/drivers/net/ethernet/qlogic/netxen/netxen_nic_hw.c
@@ -648,7 +648,7 @@ nx_p3_sre_macaddr_change(struct netxen_adapter *adapter, u8 *addr, unsigned op)
 
  mac_req = (nx_mac_req_t *)&req.words[0];
  mac_req->op = op;
- memcpy(mac_req->mac_addr, addr, 6);
+ memcpy(mac_req->mac_addr, addr, ETH_ALEN);
 
  return netxen_send_cmd_descs(adapter, (struct cmd_desc_type0 *)&req, 1);
 }
-- 
1.7.1